### DISCUSSION TITLE
next/image 사용할 때 반드시 크기를 지정해주어야 하나요?

### DISCUSSION THREAD

**Discussion by hy57in** [teamsindy20/sobok#41] [Next.js]
```tsx
<Image src="images/category-all.svg" alt="category-all" />
```
이렇게 src, alt만 지정해 놓으면 다음과 같은 에러가 뜬다.
![image](https://user-images.githubusercontent.com/60775453/131291267-d8c4b08f-b11e-40b5-9a8a-42dae6826f02.png)
크기를 반드시 지정해야 하는 불편함이 있는데...

여기서 크기란?

**Comment by hy57in** [comment]
`<Image src="images/category-all.svg" alt="category-all" layout="fill" />`
근데 이렇게 하면 빌드에러가 뜬단말이지...🤔

**Reply by gwak2837** [reply]
이건 왜 그러는지 잘 모르겠어요.. svg는 `layout="fill"`을 사용할 수 없는 걸까요?

**Reply by hy57in** [reply]
이미지 경로에 / 안써줬네요;;ㅋㅋㅋ

**Comment by gwak2837** [comment]
이미지 크기를 지정하는 방법은 여러가지 있습니다.

### 상위 HTML 태그에 의존

`next/image` 컴포넌트가 `layout=fill` 이면 해당 이미지의 가로-세로 크기는 자체적으로 크기를 가지지 않고 상위 태그에 의존하게 됩니다. 그래서 `layout=fill`을 사용하면 상위 태그의 크기가 존재해야 합니다.

### 최대 크기 명시

`next/image` 컴포넌트에 `width, height`이 있으면 해당 이미지는 명시된 크기를 가집니다. 하지만 상위 HTML 태그 크기가 이미지 크기보다 작아지면 이미지도 그에 맞춰 줄어듭니다. 즉, `width, height`는 이미지의 최대 크기라고 볼 수 있습니다.

**Reply by hy57in** [reply]
설명 감사합니다!
추가적으로 `layout="fill"`을 사용하려면 상위태그가 `position:relative`이어야 하네요
